perceval/backends/core/gitter.py

```diff
@@ -0,0 +1,325 @@
+            page += 1
+            logger.debug("Page: %i" % page)
+            message_group = self.client.message_page(self.room_id, before_id)
+
+            if not message_group:
+                fetching = False
+                continue
+
+            message_group = message_group.json()
+            for raw_message in message_group:
+
+                if str_to_datetime(raw_message['sent']).timestamp() > from_date:
+                    num_msgs += 1
+                    yield raw_message
+                else:
+                    fetching = False
+
+            before_id = message_group[0]['id']
+
+        logger.debug("Fetch process completed: %s message fetched", num_msgs)
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not support items resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Gitter item."""
+
+        return item['id']
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts and coverts the sent time of a message
+        from a Gitter item.
+
+        The timestamp is extracted from 'sent' field and
+        converted to a UNIX timestamp.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = str_to_datetime(item['sent'])
+
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Gitter item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        room = urijoin(self.group, self.room)
+
+        return GitterClient(self.api_token, self.max_items, self.archive,
+                            from_archive, self.ssl_verify, room=room)
+
+
+class GitterClient(HttpClient):
+    """Gitter API client.
+
+    Client for fetching information from the Gitter server
+    using its REST API.
+
+    :param api_token: key needed to use the API
+    :param max_items: maximum number of items per request
+    :param archive: an archive to store/read fetched data
+    :param from_archive: it tells whether to write/read the archive
+    :param ssl_verify: enable/disable SSL verification
+    :param room: room from which the message are to be fetch.
+        This contains the absolute path of the room i.e. {group}/{room}
+    """
+    AUTHORIZATION_HEADER = 'Authorization'
+    GET_MESSAGES_ENDPOINT = 'chatMessages'
+    GET_ROOMS_ENDPOINT = 'rooms'
+
+    def __init__(self, api_token, max_items=MAX_ITEMS, archive=None,
```

Comment:
The client should implement the method `sanitize_for_archive`, which is used to remove the token information from the requests before storing them to the archive (ex.: https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/gitlab.py#L684).

This method isn't present in other clients (like the ones below).
- https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/pagure.py
- https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/github.py

 An issue should be opened to check how many clients miss this method and if this is preventing to fetch archived data generated with the corresponding backends. @animeshk08 let me know if you want to open this issue, otherwise I can do it :)

Reply:
Thanks for the reference! I will open the issue:)